### PR TITLE
{bio}[goolf/1.4.10] Minimac 20140110 (REVIEW)

### PR DESCRIPTION
**Pull request by pescobar**

### CODE REVIEW

**Comment by fgeorgatos** [comment]
Test report by @fgeorgatos
**FAILED**
Build succeeded for 0 out of 1
Linux debian 6.0.10, Intel(R) Xeon(R) CPU           L5640  @ 2.27GHz, Python 2.6.6
See https://gist.github.com/28db2609298a8cb1ab3e for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/b3256837df6f296167e1 for a full test report.

File: easybuild/easyconfigs/m/Minimac/Minimac-2013.7.17-goolf-1.4.10.eb [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# This file is an EasyBuild reciPY as per https://github.com/hpcugent/easybuild
+# Author: Pablo Escobar Lopez
+# Swiss Institute of Bioinformatics
+# Biozentrum - University of Basel
+
+easyblock='MakeCp'
+
+name = 'Minimac'
+version = '2013.7.17'
+versionsuffix = '-beta'
```
Comment (boegel) [inline easybuild/easyconfigs/m/Minimac/Minimac-2013.7.17-goolf-1.4.10.eb]: why beta?

File: easybuild/easyconfigs/m/Minimac/Minimac-2013.7.17-goolf-1.4.10.eb [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# This file is an EasyBuild reciPY as per https://github.com/hpcugent/easybuild
+# Author: Pablo Escobar Lopez
+# Swiss Institute of Bioinformatics
+# Biozentrum - University of Basel
+
+easyblock='MakeCp'
+
+name = 'Minimac'
+version = '2013.7.17'
```
Comment (boegel) [inline easybuild/easyconfigs/m/Minimac/Minimac-2013.7.17-goolf-1.4.10.eb]: current tarball is from 20140110

**Comment by rjeschmi** [comment]
Test report by @rjeschmi
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux centos 6.6, Intel(R) Xeon(R) CPU           E5440  @ 2.83GHz, Python 2.6.6
See https://gist.github.com/4c0db7652130ac4b4e5a for a full test report.

File: easybuild/easyconfigs/m/Minimac/Minimac-2013.7.17-goolf-1.4.10.eb (hunk 1 above)
Comment (fgeorgatos) [inline easybuild/easyconfigs/m/Minimac/Minimac-2013.7.17-goolf-1.4.10.eb]: Hi folks,

I try to enter into Pablo's mind and I think he would have liked to propose instead:
```
version = '2013.7.17'
versionsuffix = '-a8011a4'
```

More generally, for nasty version-less applications we should adopt an ad-hoc format like:
```
version = '20130717'
versionsuffix = '-a8011a4' # or any other hash for that matter
```

The point is:
* Make it possible to sort by temporal release
* Confine the version/naming sufficiently to make the builds truly unique.

Your takes?

File: easybuild/easyconfigs/m/Minimac/Minimac-2013.7.17-goolf-1.4.10.eb (hunk 2 above)
Comment (fgeorgatos) [inline easybuild/easyconfigs/m/Minimac/Minimac-2013.7.17-goolf-1.4.10.eb]: ditto; a hash would help to confine what we are talking about...

File: easybuild/easyconfigs/m/Minimac/Minimac-2013.7.17-goolf-1.4.10.eb (hunk 1 above)
Comment (boegel) [inline easybuild/easyconfigs/m/Minimac/Minimac-2013.7.17-goolf-1.4.10.eb]: makes sense, yeah

We should be using the datestamp in the source tarball name too, but that breaks auto-downloading.

File: easybuild/easyconfigs/m/Minimac/Minimac-2013.7.17-goolf-1.4.10.eb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+ The name has two parts. The first, 'mini', refers to the modest amount of computational
+ resources it requires. The second, 'mac', is short hand for MaCH, our widely used 
+ algorithm for genotype imputation.  """
+
+toolchain = {'name': 'goolf', 'version': '1.4.10'}
+
+# the source file doesn't include version in the name so probably you should check
+# in their website if the version number and date is correct
+source_urls = ['http://www.sph.umich.edu/csg/cfuchsb/']
+sources = ['%(namelower)s.src.tgz']
```
Comment (boegel) [inline easybuild/easyconfigs/m/Minimac/Minimac-2013.7.17-goolf-1.4.10.eb]: include datestamp/version in source tarball name (breaks auto-downloading, but that's not a big issue imho)

```python
# download minimac.src.tgz from http://www.sph.umich.edu/csg/cfuchsb and rename
sources = ['%(namelower)s-%(version)s.src.tgz']
```

**Comment by boegel** [comment]
@pescobar: please review/merge https://github.com/pescobar/easybuild-easyconfigs/pull/13, so we can get this one merged in...

File: easybuild/easyconfigs/m/Minimac/Minimac-2013.7.17-goolf-1.4.10.eb (hunk 1 above)
Comment (boegel) [inline easybuild/easyconfigs/m/Minimac/Minimac-2013.7.17-goolf-1.4.10.eb]: on second thought, just specifying a datestamp as 'version' as using that in the source name should do it, no need to add a hash as a version suffix (the datestamp serves that purpose, together with the specified MD5sum being checked)

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/3dd7df38a8a0893cc536 for a full test report.

**Comment by boegel** [comment]
(finally) going in, thanks @pescobar, and @fgeorgatos, @rjeschmi for testing/reviewing

**Pull request merged**